src/poetry/packages/locker.py

```diff
@@ -291,6 +293,10 @@ def _should_write(self, lock: TOMLDocument) -> bool:
 
     def _write_lock_data(self, data: TOMLDocument) -> None:
         lockfile = TOMLFile(self.lock)
+        if self.lock.exists():
+            # get original line ending.
+            with contextlib.suppress(TOMLError):
+                lockfile.read()
```

Comment:
```suggestion
        if self.lock.exists():
            # The following code is roughly equivalent to
            # • lockfile = TOMLFile(self.lock)
            # • lockfile.read()
            # • lockfile.write(data)
            # However, lockfile.read() takes more than half a second even
            # for a modestly sized project like Poetry itself and the only reason
            # for reading the lockfile is to determine the line endings. Thus,
            # we do that part for ourselves here, which only takes about 10 ms.

            # get original line endings
            with open(self.lock, encoding="utf-8", newline="") as f:
                line = f.readline()
            linesep = "\r\n" if line.endswith("\r\n") else "\n"

            # enforce original line endings
            content = data.as_string()
            if linesep == "\n":
                content = content.replace("\r\n", "\n")
            elif linesep == "\r\n":
                content = re.sub(r"(?<!\r)\n", "\r\n", content)
            with open(self.lock, "w", encoding="utf-8", newline="") as f:
                f.write(content)

        else:
            lockfile = TOMLFile(self.lock)
            lockfile.write(data)
```

Reply:
or, abandon the notion that a particular formatting-style for lockfiles is important and write them with tomli-w - which always uses `\n` exclusively and therefore can never flip-flop.

I realise that I've suggested this in the past and I guess we likely continue to disagree on the merits of this - but I continue to favour eliminating tomlkit wherever it is possible to do so.